src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -92,5 +107,17 @@ def interpreter_constraints(self) -> Tuple[str, ...]:
         return tuple(self.options.interpreter_constraints)
 
     @property
-    def entry_point(self) -> Optional[str]:
-        return cast(Optional[str], self.options.entry_point)
+    def main(self) -> MainSpecification:
+        is_default_console_script = self.options.is_default("console_script")
+        is_default_entry_point = self.options.is_default("entry_point")
+        if not is_default_console_script and not is_default_entry_point:
+            raise OptionsError(
+                f"Both --console-script={self.options.console_script} and "
+                f"--entry-point={self.options.entry_point} are configured but these options are "
+                f"mutually exclusive. Pick one."
+            )
```

Comment:
It'd be helpful to include `self.scope`: `f"[{self.scope}].console_script` and `f"[{self.scope}].entry_point`. 

We (Benjy, myself, and I think Stu) have been using the `[scope].opt` style instead of `--scope-opt` recently in docs because it's less ambiguous and most of these options get specified in `pants.toml` rather than the CLI anyways.

Reply:
Also, consider "Please pick one." :)

Reply:
Hrm, ok. I'll ad hoc that here but instead of "we have been pattern" it'd be better to encode that in a Subsystem helper. I'll maybe follow up with that.

And sure, but there is no please about it so this sounds strange to me. You have to pick one or stand dead in the water since we raised.

Reply:
Yeah, the subsystem helper ideally should be able to read the rank of the option and display the option in the form you specified it. That would be ideal I think.